libvmi/driver/kvm/kvm.c

```diff
@@ -277,6 +285,31 @@ parse_seg_reg_value(
     return (reg_t) strtoll(ptr, (char **) NULL, 16);
 }
 
+static addr_t
+parse_mtree(char *mtree_output)
+{
+    char *ptr;
+
+    if(NULL == (ptr = strstr(mtree_output, "alias ram-above-4g")))
+    goto error;
+
+    for(int i = 0; i <= 50;i++) {
+      if(ptr[0] == '-') {
+        ptr++;
+        goto success;
+      }
+      ptr--;
+    }
+
+    goto error;
+
+    error:
+    return 0;
+
+    success:
+    return (addr_t) strtoll(ptr, (char **) NULL, 16) + 1;
```

Comment:
Why do you return `strtoll` + 1 ?

Reply:
So the examples don't go trying to read one byte past the reported end of ram-above-4g and throwing an error message. (Note the end of alias-above-4g is always something like 000000013fffffff, so adding one will give us a nice round number)